napari/_qt/layer_controls/qt_layer_controls_container.py

```diff
@@ -108,6 +111,16 @@ def __init__(self, viewer):
         self.viewer.layers.events.inserted.connect(self._add)
         self.viewer.layers.events.removed.connect(self._remove)
         viewer.layers.selection.events.active.connect(self._display)
+        viewer.dims.events.ndisplay.connect(self._on_ndisplay_change)
+
+        # TODO: I think this is needed if viewer is initialized with ndisplay=3,
+        # but maybe we should do this differently.
+        self._on_ndisplay_change(Event('ndisplay', value=viewer.dims.ndisplay))
+
+    def _on_ndisplay_change(self, event):
+        for widget in self.widgets.values():
+            if widget is not self.empty_widget:
+                widget.ndisplay = event.value
```

Comment:
This is kind ahow we handle `VispyLayer`s in the backend (calling `reset()`), so it makes sense to me. Not sure we need the event actually, since you can simply do:

```py
self._on_display_change()
```

and then 

```py
    def _on_ndisplay_change(self):
        for widget in self.widgets.values():
            if widget is not self.empty_widget:
                widget.ndisplay = viewer.dims.ndisplay
```

Just like we handle all these things in `VispyLayer`s.

Reply:
Good to know. I actually realized that `_on_ndisplay_change` will never do anything at this point since `self.widgets` only contains `self.empty_widget`, so I just deleted this.